content/polyhedra/functions.ts

```diff
@@ -21,36 +21,175 @@ import './components/folding';
 import './components/anibutton';
 
 
-export function angles($step: Step) {
-  const totals = [360, 540];
-  const $buttons = $step.$$('x-anibutton') as Anibutton[];
-  let overlap = [false, false];
+// Attempted fix for https://github.com/mathigon/textbooks/issues/8
+//
+// - Uses computeChangedAngles() during model.watch() to figure out which points
+// in a GeoPad have moved, and which angles are expected to be adjusted.
+//
+// - Uses computeGeometry() to determine the angles of the polygon, rounded to integers
+// in such a way that only angles adjacent to the user's action (e.g., dragging a point)
+// will be affected by the need to add +1 or -1 to deal with rounding error.
+//
+
+// computeGeometry()
+// Compute the necessary angles and integer-rounded values for these values.
+// If the sum of the integer-rounded angles is not the expected sum (e.g., 360 for a quadrilateral),
+// then choose an angle to add/subtract the rounding error to create the correct sum.
+// The angle to be adjusted will be chosen from the set of adjustableAngleNames, if non-empty,
+// and will pointNames[0] otherwise.
+//
+// The result of computeGeometry will be a geometry object containing the calculated
+// angles and their rounded forms.
+//
+// This function also computes a .overlap boolean which indicates whether
+// there are overlapping edges
+//
+
+function computeGeometry(model, pointNames, adjustableAngleNames): object {
```

Comment:
Don't return just an `object`, but instead create an interface that contains all the required keys. However, it might be even better to just return an array of {exact, rounded} objects, in the same other of vertices as the polygon. The sum/roundedSum/overlap values can be extracted easily when they are required.